config/externalname.go

```diff
@@ -29,6 +29,9 @@ var TerraformPluginFrameworkExternalNameConfigs = map[string]config.ExternalName
 	// terraform-plugin-framework
 	"aws_appconfig_environment": appConfigEnvironment(),
 
+	// AWS Batch job queue can be imported using the name
+	"aws_batch_job_queue": config.NameAsIdentifier,
```

Comment:
In the terraform document, I see that the resource will be imported with arn in the [import](https://registry.terraform.io/providers/hashicorp/aws/5.68.0/docs/resources/batch_job_queue#import) section. Therefore, in order to use the [NameAsIdentifier](https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#case-1-name-as-identifier) configuration, the arn of the resource and the name parameters must be the same. As I see they are not same so here we can use [IdentifierFromProvider](https://github.com/crossplane/upjet/blob/main/docs/adding-new-resource.md#case-3-random-identifier-from-provider) or TemplatedStringAsIdentifier, please try below:

```suggestion
	"aws_batch_job_queue": config.TemplatedStringAsIdentifier("name", "arn:aws:batch:{{ .setup.configuration.region }}:{{ .client_metadata.account_id }}:job-queue/{{ .external_name }}"),
```